src/source_map_utils.ts

```diff
@@ -109,48 +109,37 @@ export function sourceMapTextToGenerator(sourceMapText: string): SourceMapGenera
   return SourceMapGenerator.fromSourceMap(sourceMapTextToConsumer(sourceMapText));
 }
 
+/**
+ * A position in a source map. All offsets are zero-based.
+ */
 export interface SourcePosition {
-  // 0 based
+  /** 0 based */
   column: number;
-  // 0 based
+  /** 0 based */
   line: number;
-  // 0 based
+  /** 0 based full offset in the file. */
   position: number;
 }
 
 export interface SourceMapper {
+  /**
+   * Logically shift all source positions by `offset`.
+   *
+   * This method is useful if code has to prepend addition text to the generated output after source
+   * mappings have already been generated. The source maps are then transparently adjusted during
```

Comment:
This makes me think we got the layering wrong.  We wanted to move all of this to using transformers, so perhaps this new code could try?

Reply:
Yeah, but this new code is affecting how we annotate, so for this we'd need to move annotate, which is kind of all of tsickle. I considered it, and we really have to do it sooner than later, but it's a bunch of work.

Reply:
I think to incrementally move code out of annotate, we will have to make it export data in a form such that the transformer wrapper can use it.  In this case you could export the list of things you want imported, and then the transformer that calls annotate would use that list to generate source nodes for the imports.

Maybe it's not worth worrying about right now, but (1) at our current rate the transformer thing will never happen, and (2) this is maybe an easy one to do it for because you genuinely don't want these imports in the source text (which is why you have to monkey with the source map).

Reply:
I had the same feeling, and looked into what it'd take. The problem is twofold: we'd need to pipe this "extra prefix" through several layers of code, back up to `transformer_sourcemap`. That's a bit nasty, but more importantly, we still need to make sure the effective source map is correct. If we prepended the text in the transformer before parsing, we'd still need to handle the offset as we do now. If we hand the prefix further up, then we'll break source maps because we prepend after those are generated.

I think folding this into the source mapper is OK, it's the abstraction we use to bridge from the text format used in annotate to the transformer mode, so it's the right place.